docs/enhancements/metallb-operator.md

```diff
@@ -0,0 +1,205 @@
+(MetalLB CR) and the Service definition look like in the case of the apiserver:
+
+```yaml
+---
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: metallb-api-vip
+  namespace: metallb
+spec:
+  protocol: layer2
+  autoAssign: false
+  addresses:
+    # We will be discussing this further down in the enhancement
+    - $METALLB_IP/32
+---
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    # This is used by MetalLB to know from which address
+    # pool it will take an IP for this service
+    metallb.universe.tf/address-pool: metallb-api-vip
+  name: metallb-api
+  namespace: openshift-kube-apiserver
+spec:
+  ports:
+  - name: http
+    protocol: TCP
+    port: 6443
+    targetPort: 6443
+  selector:
+    app: openshift-kube-apiserver
+  type: LoadBalancer
+```
+
+## Proposal
+
+Adding an operator to Assisted Installer is a fairly simple task. This operator, however, requires a
+bit more input for Assisted Installer to be able to configure it properly. Specifcally, Assisted
+Installer needs to be able to answer the following questions:
+
+- Which is the *internal network*?
+- What is the IP on the **external network** to use for the API VIP?
+- What is the IP on the **external network** to use for the Ingress?
+- What routes (if any) should be used for the **external** API and Ingress endpoints?
+
+### Identifying Internal Network
+
+Assisted installer already has mechanims to know which is the internal network. Internally this
+network is called the `MachineNetwork`. In case of SNO, the `MachineNetwork` is required when defining
+the ACI. In the case of a multinode deployment, it is inferred from the API VIP whereas in the case
+of dual stack this information is required the same way as in the SNO case.
+
+Regardless, no changes should be required in assisted installer to identify the internal network.
+
+### Identifying External Network
+
+This enhancement proposes requesting the user for the following information in order to prepare the
+cluster for relocation:
+
+- External API VIP: This is an IP on the **external** network
+- External Ingress IP: This is an IP on the **external** network
+- External base domain: A domain to use for the **external** traffic (optional).
+
+Unlike for the **internal** network, this enhancement proposes to always infer the subnet of the
+external network from the External API VIP and the External Ingress IP provided by the user. There
+are utilities in Assisted Installer to do this already.
+
+Some requirements:
+
+- The **external** API VIP and Ingress IP must be on the same subnet.
+- The host must be connected to this network and the network must be routable
+- These 2 IPs must be optional. The IPAddressPool and Service won't be configured if this information is not provided
+- Either both are IPs are provided or none.
+- The *internal* network must be configured as a static network to prevent the MachineNetwork to
```

Comment:
Why it should be static? I mean there is a requirement that ip of the nodes can't change as etcd will break so why it is required to be "static" here?

Reply:
This is because we don't want this IP to change during the relocation. As you said, everything will break if this IP changes. We can prevent this by having this internal IP as static and isolated. That is, the internal network is not expected to be connected to anything.

This is how we achieved relocation for ZTPFW. The big difference is that, in ZTPFW, there's a specific IP hardcoded. In this enhancement, the idea is to let the user's pick the IP they want.

Hope it makes sense

Reply:
The question is if we are going to block installation if user don't set static network or we will allow to spawn metalb when ever user wants.
I think we should just allow spawning metalb operator without checking if static network was configured or not

Reply:
In that case we can decouple "internal network" configuration from metalb and say that we want to allow installing metalb from assisted on user demand. This will allow us to minimize validations that will be required.

Reply:
> I think we should just allow spawning metalb operator without checking if static network was configured or not

mmh, not sure I agree with this. The reason is that the whole point of this feature is to build clusters that can be relocated. If the internal network is not static, the cluster will break if/once it is relocated. We would allow the user to build a cluster that will break on the first move.

We could argue that installing MetalLB should not mean "relocatable" but the configuration we are applying is based on that assumption

Reply:
I think we should not argue, assisted should not support something specific for "relocatable", it should support "generic" option and if you want to make "relocatable" just apply relevant configuration to make it such. 
I mean that one day it will be in SAAS and people will want it and they are not going to relocate their cluster, they just want to use this functionality, why not to give to them. 
IMO if user needs static network he should configure it , if he doesn't why to block him.

Reply:
In case we are talking only about "relocatable" (not sure i understand why we need to implement this very specific scenario and why external tool is not enough for this as it can apply relevant configuration) we should change name of the enhancement and rephrase it.
Though we were already talked about MetalLb for agent capi and it is already used there (not sure how we apply it). I think will nicer to support all the options and make it generic as we will move there for sure one day